### DISCUSSION TITLE
corfu completion doesnt seem to work in org elisp blocks

### DISCUSSION THREAD

**Discussion by rileyrg** [minad/corfu#418] [Q&A] [answered]
eg inside  here no completion happens : global corfu mode is on.

```
**** corfu
#+begin_src emacs-lisp
  (use-package corfu
    ;; Optional customizations
    :custom
    ;; (corfu-cycle t)                ;; Enable cycling for `corfu-next/previous'
    (corfu-auto t)                 ;; Enable auto completion
    ;; (corfu-separator ?\s)          ;; Orderless field separator
    ;; (corfu-quit-at-boundary nil)   ;; Never quit at completion boundary
    ;; (corfu-quit-no-match nil)      ;; Never quit, even if there is no match
    ;; (corfu-preview-current nil)    ;; Disable current candidate preview
    ;; (corfu-preselect 'prompt)      ;; Preselect the prompt
    ;; (corfu-on-exact-match nil)     ;; Configure handling of exact matches
    ;; (corfu-scroll-margin 5)        ;; Use scroll margin

    ;; Enable Corfu only for certain modes.
    ;; :hook ((prog-mode . corfu-mode)
    ;;        (shell-mode . corfu-mode)
    ;;        (eshell-mode . corfu-mode))

    ;; Recommended: Enable Corfu globally.  This is recommended since Dabbrev can
    ;; be used globally (M-/).  See also the customization variable
    ;; `global-corfu-modes' to exclude certain modes.
    :init
    (global-corfu-mode))


  ;; A few more useful configurations...
  (use-package emacs
    :init
    ;; TAB cycle if there are only few candidates
    (setq completion-cycle-threshold 3)

    ;; Emacs 28: Hide commands in M-x which do not apply to the current mode.
    ;; Corfu commands are hidden, since they are not supposed to be used via M-x.
    ;; (setq read-extended-command-predicate
    ;;       #'command-completion-default-include-p)

    ;; Enable indentation+completion using the TAB key.
    ;; `completion-at-point' is often bound to M-TAB.
    (setq tab-always-indent 'complete))
#+end_src

```

**Comment by minad** [comment, marked as answer]
Yes, of course not. You need a special capf for this. See cape-elisp-block.

**Reply by rileyrg** [reply]
Why of course not?  It just works in normal el files.  Company just figures it out. But thanks for the pointer. I was just trying out Corfu again and will look into this cape thing.

**Reply by minad** [reply]
> Why of course not? It just works in normal el files.

The builtin Elisp Capf does not support Org files. How should it know about such code blocks? Also it is not installed as a backend by default for arbitrary files.

> Company just figures it out.

"just" - I am not entirely sure know it works, but I assume that the extra Company backend comes with special support for source blocks like `cape-elisp-block`, see https://github.com/company-mode/company-mode/blob/master/company-elisp.el. Note that Company comes with a bunch of separate backends. Maybe do you also have some special configuration?

**Comment by rileyrg** [comment]
I'm not saying it should ;) Just that it is possible. I have a little
snippet in a global "get help" function I wrote which does just this for
relevant help facilities

      (if (or (derived-mode-p  'emacs-lisp-mode) (and (eq
   major-mode 'org-mode) (string= "emacs-lisp" (car
(org-babel-get-src-block-info)))))

>           (rgr/emacs-lisp-help)



Anyway, more than happy to use CAPE if  thats the approach : consult my
favourite package and I thought I'd try corfu again.

regards

r.
